### PR TITLE
Add test for superfluous witness record in deserialization

### PR DESCRIPTION
**Pull request by instagibbs**
Adds coverage for changed behavior in https://github.com/bitcoin/bitcoin/pull/14039

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Tests are failing with 

```
AssertionError: [node 0] Expected message "bad-txns-vin-empty" does not partially match log:
 - 2019-04-25T19:05:50.370758Z received: tx (119 bytes) peer=0
 - 2019-04-25T19:05:50.370863Z 
 - 
 - ************************
 - EXCEPTION: NSt8ios_base7failureB5cxx11E       
 - Superfluous witness record: iostream error       
 - bitcoin in ProcessMessages()

**Comment by instagibbs** [comment]
wait, what? I didn't touch this test:
`64/119 - p2p_invalid_tx.py failed, Duration: 1 s`

**Comment by MarcoFalke** [comment]
Could you include the following diff:

```diff
diff --git a/test/functional/data/invalid_txs.py b/test/functional/data/invalid_txs.py
index 02deae92f3..80d8d66f24 100644
--- a/test/functional/data/invalid_txs.py
+++ b/test/functional/data/invalid_txs.py
@@ -68,7 +68,7 @@ class OutputMissing(BadTxTemplate):
 
 
 class InputMissing(BadTxTemplate):
-    reject_reason = "bad-txns-vin-empty"
+    reject_reason = "Superfluous witness record"
     expect_disconnect = False
 
     def get_tx(self):

**Comment by instagibbs** [comment]
@MarcoFalke I tried a slightly different track by using the fact that 0-vin, 0-vout have input sizes checked first, and kept that test case, then also kept my new test for the actual test we're checking for. It seemed weird to me to simply hope that that type of transaction would trigger this case.

File: test/functional/feature_block.py [hunk 1]
```diff
@@ -146,20 +146,6 @@ def run_test(self):
             badtx = template.get_tx()
             if TxTemplate != invalid_txs.InputMissing:
                 self.sign_tx(badtx, attempt_spend_tx)
-            else:
-                # Segwit is active in regtest at this point, so to deserialize a
-                # transaction without any inputs correctly, we set the outputs
-                # to an empty list. This is a hack, as the serialization of an
-                # empty list of outputs is deserialized as flags==0 and thus
-                # deserialization of the outputs is skipped.
```
Comment (MarcoFalke) [inline test/functional/feature_block.py]: Please move this comment to `data/invalid_txs.py`, as it is not clear why the tx there is completely "empty"

Comment (instagibbs) [reply test/functional/feature_block.py]: I wrote something else in its place. I found the previous text a little baffling to be honest. Willing to take heavy edits on it :)

**Comment by MarcoFalke** [comment]
utACK cc556e4a30b4a32eab6722f590489d89b2875de3

**Pull request merged**